client/extensions/zoninator/components/data/query-feed/index.jsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestFeed } from '../../../state/feeds/actions';
+
+class QueryFeed extends PureComponent {
+
+	static propTypes = {
+		siteId: PropTypes.number,
+		zoneId: PropTypes.number,
+	};
+
+	componentWillMount() {
+		this.requestFeed( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! nextProps.siteId || ! nextProps.zoneId ) {
+			return;
+		}
+
+		this.requestFeed( nextProps );
+	}
+
+	requestFeed( props ) {
+		const { siteId, zoneId } = props;
+
+		if (
+			! siteId ||
+			! zoneId ||
+			siteId === this.state.siteId ||
```

Comment:
This component doesn't have any state, does it? What if the `siteId` is the same but `zoneId` has changed, or vice versa? Would we still want to query for data in either of those scenarios?

This check could be done in `componentWillReceiveProps` instead:

```
if ( ! nextProps.siteId || ! nextProps.zoneId || ( this.props.siteId === nextProps.siteId && this.props.zoneId === nextProps.zoneId )  ) {
	return;
}
```